File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/LetterAttachmentDB/LetterAttachmentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LetterAttachmentRow } from "./LetterAttachmentRow";

export interface LetterAttachmentColumns {
    Title: Column<LetterAttachmentRow>;
    AttachmentFile: Column<LetterAttachmentRow>;
}

export class LetterAttachmentColumns extends ColumnsBase<LetterAttachmentRow> {
    static readonly columnsKey = 'LetterAttachmentDB.LetterAttachment';
    static readonly Fields = fieldsProxy<LetterAttachmentColumns>();
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/LetterAttachmentDB/LetterAttachment/LetterAttachmentColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.LetterAttachmentDB.Columns;

[ColumnsScript("LetterAttachmentDB.LetterAttachment")]
[BasedOnRow(typeof(LetterAttachmentRow), CheckNames = true)]
public class LetterAttachmentColumns
{

    [EditLink]
    public string Title { get; set; }
    public string AttachmentFile { get; set; }

}